### PR TITLE
fix read buffered channel

### PR DESCRIPTION
**Pull request by snowplayfire**
/kind bug

### CODE REVIEW

**Comment by snowplayfire** [comment]
@xiang90 I use it and find the bug, please help to review the PR, thanks

File: sql.go [hunk 1]
```diff
@@ -49,21 +49,33 @@ func (ds *SQLDBSyncer) Sync(ctx context.Context) error {
...
+FOR:
+	for {
+		select {
+		case r, ok := <-rc:
+			// ok == false means that buffered channel rc is closed;
+			// if rc is closed and len(r.Kvs) > 0, do process;
+			// if rc is closed and len(r.Kvs) == 0, break for loop because channel is empty and closed;
+			// if rc is open, do process. maybe len(r.Kvs) == 0.
+			if !ok && len(r.Kvs) == 0 {
+				break FOR
```
Comment (xiang90) [inline sql.go]: can we use a flag to break out the for loop instead? I really dislike the jump tag.

File: sql.go [hunk 2]
```diff
@@ -49,21 +49,33 @@ func (ds *SQLDBSyncer) Sync(ctx context.Context) error {
...
 
 	rc, errc := ds.Syncer.SyncBase(ctx)
-	select {
-	case r := <-rc:
-		for _, kv := range r.Kvs {
-			if ds.NoOverwrite {
-				if _, err := stmt.Exec(kv.Key, clientv3.EventTypePut.String(), kv.Value, kv.ModRevision); err != nil {
+FOR:
+	for {
+		select {
```
Comment (xiang90) [inline sql.go:57]: basically adding a flag check here. and get rid of the FOR tag.

Comment (snowplayfire) [reply sql.go:57]: Thanks for your suggestion, I have updated it.

**Pull request merged**